crates/red_knot_python_semantic/src/types.rs

```diff
@@ -6102,6 +6095,85 @@ impl<'db> FunctionType<'db> {
     }
 }
 
+// This is a separate block for now to avoid making `to_overloaded` a salsa query. This is a
+// temporary workaround until salsa supports returning `Option<&T>` from a tracked query.
+impl<'db> FunctionType<'db> {
+    /// Returns `self` as [`OverloadedFunction`] if it is overloaded, [`None`] otherwise.
+    fn to_overloaded(self, db: &'db dyn Db) -> Option<&'db OverloadedFunction<'db>> {
```

Comment:
It would be great to document the limitations that the method only returns overloads defined *before* `self` as this may not be immediately obvious from the implementation

Reply:
Added a note in [`734a708` (#17585)](https://github.com/astral-sh/ruff/pull/17585/commits/734a708fb407b4a1c90c1e2c4e8becea97cedd5d).